src/components/Assemblies/CabinetRole.svelte

```diff
@@ -1,21 +1,67 @@
 <script lang="ts">
+	import type { AssemblyMember } from '$lib/datasheets/assembly-member';
 	import type { CabinetSeat } from './shared';
 
 	export let cabinets: CabinetSeat[];
 	export let role: CabinetSeat['role'];
+
+	const show = (event: MouseEvent, member: AssemblyMember) => {
+		const memberName = `${member.firstname} ${member.lastname}`;
+		const tooltipElem = document.getElementById('tooltip');
+
+		if (tooltipElem) {
+			tooltipElem.setAttribute(
+				'style',
+				`left: ${event.layerX}px; top: ${event.layerY + 20}px; display: flex;`
+			);
+		}
+
+		const memberElem = document.getElementById('member-name');
+		const roleElem = document.getElementById('role');
+
+		if (memberElem) {
+			memberElem.textContent = memberName;
+		}
+
+		if (roleElem) {
+			roleElem.textContent = member.assemblyRole.role;
+		}
+	};
+
+	const hide = () => {
+		const tooltipElem = document.getElementById('tooltip');
+
+		if (tooltipElem) {
+			tooltipElem.setAttribute('style', `display: none;`);
+		}
+	};
 </script>
 
 <div class="role">
 	<p class="heading-compact-01">{role}</p>
 	<div class="group-dot">
 		{#each cabinets.find((c) => role === c.role)?.parties || [] as party}
-			{#each Array.from({ length: party.count || 0 }, (_, i) => i) as _}
-				<span class="dot" style="background-color: {party.color || '#8D8D8D'};"></span>
+			{#each party?.members ?? [] as member}
+				<!-- svelte-ignore a11y-no-static-element-interactions -->
+				<span
```

Comment:
`span` was designed to be "an inline container used to mark up a part of a text, or a part of a document" [(W3C)](https://www.w3schools.com/tags/tag_span.asp). So it is not suitable for this case, even though it works. Please change to `div`.

HTML was designed to be [semantic](https://www.w3schools.com/html/html5_semantic_elements.asp). Browser, SEO, screen reader (for blind people), and library might treat each HTML tag differently, we should use it accordingly.

Reply:
Thank you, fixed.